pkg/pipeline/ChartService.go

```diff
@@ -129,6 +130,11 @@ type ChartService interface {
 	DeploymentTemplateValidate(templatejson interface{}, chartRefId int) (bool, error)
 	JsonSchemaExtractFromFile(chartRefId int) (map[string]interface{}, error)
 	GetSchemaAndReadmeForTemplateByChartRefId(chartRefId int) (schema []byte, readme []byte, err error)
+	ExtractChartIfMissing(chartData []byte, refChartDir string, location string) (*ChartDataInfo, error)
+	CheckChartExists(chartRefId int) error
+	GetChartLocation(chartDir string, fileName string) string
+	ValidateFileUploaded(fileName string) error
+	ReadChartYamlForLocation(chartDir string, fileName string) (string, string, error)
```

Comment:
can update the names of the output data variables here instead of only showing the type

Reply:
change the name of the function also, focus not on how but what. It is more like GetChartNameAndVersionFromLocation

Reply:
most of the functions are following the above format

Reply:
main action is fetching chart metadat and not reading file, if tomorrow we start reading it from some other file its name should not change

Reply:
this is pending